utils/generic.py

```diff
@@ -308,3 +308,184 @@ def annotation_type_in_vcf_info(t: Any) -> bool:
+            """
+            Converts an allele from a genotype into a one-hot-encoded vector of n_alleles length.
+            E.g. gt = 0/1
+            => allele_to_vector(gt, 0, 3) => [1, 0, 0]
+            => allele_to_vector(gt, 1, 3) => [0, 1, 0]
+
+            :param CallExpression gt: Genotype
+            :param int allele_index: Allele index
+            :param int n_alleles: Number of alleles at the site
+            :return: One-hot-encoded genotype-allele array
+            :rtype: ArrayExpression
+            """
+            return hl.range(0,n_alleles).map(lambda i: hl.cond(gt[allele_index]==i, 1, 0))
+
+        def get_phased_parent_gt(gt: hl.expr.CallExpression, transmitted_allele_index: int):
+            """
+            Given a genotype and which allele was transmitted to the offspring, returns the parent phased genotype.
+
+            :param CallExpression gt: Parent genotype
+            :param int transmitted_allele_index: index of transmitted allele (0 or 1)
+            :return: Phased parent genotype
+            :rtype: CallExpression
+            """
+            return hl.call(
+                gt[transmitted_allele_index],
+                hl.cond(
+                    transmitted_allele_index == 0,
+                    gt[1],
+                    gt[0]
+                ),
+                phased=True
+            )
+
+        proband_v = allele_to_vector(proband_gt, 0, n_alleles) + allele_to_vector(proband_gt, 1, n_alleles)
+        father_v = gt_to_vectors(father_gt, n_alleles)
+        mother_v = gt_to_vectors(mother_gt, n_alleles)
+
+        combinations = hl.flatmap(
+            lambda f:
+            hl.zip_with_index(mother_v)
+                .filter(lambda m: m[1] + f[1] == proband_v)
+                .map(lambda m: hl.struct(m=m[0], f=f[0])),
+            hl.zip_with_index(father_v)
+        )
+
+        return hl.cond(hl.len(combinations) == 1,
+                       hl.array([
+                           hl.call(father_gt[combinations[0].f], mother_gt[combinations[0].m], phased=True),
+                           get_phased_parent_gt(father_gt, combinations[0].f),
+                           get_phased_parent_gt(mother_gt, combinations[0].m)
+                       ]),
+                       hl.array([proband_gt, father_gt, mother_gt])
+                       )
+
+    if not (proband_call._indices.source == father_call._indices.source and proband_call._indices.source == mother_call._indices.source):
+        raise ValueError("All CallExpr (proband, mother, father) must originate from the  same trio MatrixTable.")
+
+
+    tm = proband_call._indices.source
+    tm = tm.annotate_entries(
+        __phased_GT=phase_trio_genotypes(
+            proband_call,
+            father_call,
+            mother_call,
+            hl.len(tm.alleles)
+        )
+    )
+
+    tm = tm.select_entries(
+        proband_entry=hl.struct(
+            **tm.proband_entry,
```

Comment:
Not really sure if there's a point in making `proband_call` etc general, if you're assuming a `proband_entry` field exists here anyway (at that point, why not just take a single TrioMatrix, with those assumptions asserted).

Reply:
Fair point. The goal was more to be able to choose which genotype field you'd want to use. But I'm now leaning towards having `trio_matrix: hl.MatrixTable` and `genotype_field: str` as args to specify this.